src/ILOManager.sol

```diff
@@ -173,44 +169,45 @@ contract ILOManager is IILOManager, Ownable, Initializable {
         ILO_POOL_IMPLEMENTATION = iloPoolImplementation;
     }
 
-    function transferAdminProject(address admin, address uniV3Pool) external override onlyProjectAdmin(uniV3Pool) {
-        Project storage _project = _cachedProject[uniV3Pool];
-        _project.admin = admin;
-        emit ProjectAdminChanged(uniV3Pool, msg.sender, admin);
+    function transferAdminProject(address admin, string calldata projectId) external override onlyProjectAdmin(projectId) {
+        _projects[projectId].admin = admin;
+        emit ProjectAdminChanged(projectId, msg.sender, admin);
     }
 
     function setDefaultDeadlineOffset(uint64 defaultDeadlineOffset) external override onlyOwner() {
         emit DefaultDeadlineOffsetChanged(owner(), DEFAULT_DEADLINE_OFFSET, defaultDeadlineOffset);
         DEFAULT_DEADLINE_OFFSET = defaultDeadlineOffset;
     }
 
-    function setRefundDeadlineForProject(address uniV3Pool, uint64 refundDeadline) external override onlyOwner() {
-        Project storage _project = _cachedProject[uniV3Pool];
-        emit RefundDeadlineChanged(uniV3Pool, _project.refundDeadline, refundDeadline);
+    function setRefundDeadlineForProject(string calldata projectId, uint64 refundDeadline) external override onlyOwner() {
+        Project storage _project = _projects[projectId];
+        emit RefundDeadlineChanged(projectId, _project.refundDeadline, refundDeadline);
         _project.refundDeadline = refundDeadline;
     }
 
     /// @inheritdoc IILOManager
-    function launch(address uniV3PoolAddress) external override {
-        require(block.timestamp > _cachedProject[uniV3PoolAddress].launchTime, "LT");
-        (uint160 sqrtPriceX96, , , , , , ) = IUniswapV3Pool(uniV3PoolAddress).slot0();
-        require(_cachedProject[uniV3PoolAddress].initialPoolPriceX96 == sqrtPriceX96, "UV3P");
-        address[] memory initializedPools = _initializedILOPools[uniV3PoolAddress];
-        require(initializedPools.length > 0, "NP");
-        for (uint256 i = 0; i < initializedPools.length; i++) {
-            IILOPool(initializedPools[i]).launch();
+    function launch(string calldata projectId, address saleToken) external override {
+        require(block.timestamp > _projects[projectId].launchTime, "LT");
+        require(msg.sender == _projects[projectId].admin, "UA");
```

Comment:
Why don't we use `onlyProjectAdmin` here?

Reply:
`require(msg.sender == _projects[projectId].admin, "UA");`